core/domain/exp_services.py

```diff
@@ -2117,3 +2117,35 @@ def delete_state_id_mapping_model_for_exploration(
     exp_versions = range(1, exploration_version + 1)
     exp_models.StateIdMappingModel.delete_state_id_mapping_models(
         exploration_id, exp_versions)
+
+
+def find_all_values_for_key(key, dictionary):
```

Comment:
Why is this in exp_services? It doesn't have anything to do with explorations; it should probably be in utils.

It is also very weird that the parameter called "dictionary" can be a dict or a list. This doesn't make sense, and lists don't have keys in any case.

There are also no tests for this function. This function has complex logic and should be covered by comprehensive unit tests.

Reply:
Done! PTAL.